src/main/java/bio/terra/app/controller/ConfigsApiController.java

```diff
@@ -0,0 +1,148 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+
+import java.util.Optional;
+
+@Controller
+@Api(tags = {"configs"})
+public class ConfigsApiController implements ConfigsApi {
+
+    private Logger logger = LoggerFactory.getLogger(ConfigsApiController.class);
+
+    private final ObjectMapper objectMapper;
+    private final HttpServletRequest request;
+    private final IamService iamService;
+    private final PolicyMemberValidator policyMemberValidator;
+    private final AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+    private final ConfigurationService configurationService;
+    private final AssetModelValidator assetModelValidator;
+    private final ApplicationConfiguration appConfig;
+
+    @Autowired
+    public ConfigsApiController(
+        ObjectMapper objectMapper,
+        HttpServletRequest request,
+        IamService iamService,
+        ApplicationConfiguration appConfig,
+        PolicyMemberValidator policyMemberValidator,
+        AuthenticatedUserRequestFactory authenticatedUserRequestFactory,
+        ConfigurationService configurationService,
+        AssetModelValidator assetModelValidator
+    ) {
+        this.objectMapper = objectMapper;
+        this.request = request;
+        this.iamService = iamService;
+        this.appConfig = appConfig;
+        this.policyMemberValidator = policyMemberValidator;
+        this.authenticatedUserRequestFactory = authenticatedUserRequestFactory;
+        this.configurationService = configurationService;
+        this.assetModelValidator = assetModelValidator;
+    }
+
+    @InitBinder
+    protected void initBinder(final WebDataBinder binder) {
+        binder.addValidators(policyMemberValidator);
+        binder.addValidators(assetModelValidator);
+    }
+
+    @Override
+    public Optional<ObjectMapper> getObjectMapper() {
+        return Optional.ofNullable(objectMapper);
+    }
+
+    @Override
+    public Optional<HttpServletRequest> getRequest() {
+        return Optional.ofNullable(request);
+    }
+
+    private AuthenticatedUserRequest getAuthenticatedInfo() {
+        return authenticatedUserRequestFactory.from(request);
+    }
+
+    @Override
+    public ResponseEntity<ConfigModel> getConfig(@PathVariable("name") String name) {
+        iamService.verifyAuthorization(
+            getAuthenticatedInfo(),
+            IamResourceType.DATAREPO,
+            appConfig.getResourceId(),
+            IamAction.CONFIGURE);
+        ConfigModel configModel = configurationService.getConfig(name);
+        return new ResponseEntity<>(configModel, HttpStatus.OK);
```

Comment:
This form is fine (and I'm sure you just copied this over) but I usually use the static methods on `ResponseEntity`
```suggestion
        return ResponseEntity.ok(configurationService.getConfig(name));
```